Colorize text pointed to by carats

kartikmawani opened the pull request:
"Fixes #911 "
This will colorize the carat and the text pointed to by the carat both in red when printing errors

rbalicki2 commented on crates/common_lang_types/src/text_with_carats.rs:
```diff
@@ -589,4 +605,21 @@ mod test {
+
+        colored::control::unset_override();
+
+        assert!(output.contains("\u{1b}[91m"));
```
could you assert that it exactly matches the expected text, please?

rbalicki2 commented on crates/common_lang_types/src/text_with_carats.rs:
```diff
@@ -589,4 +605,21 @@ mod test {
+        )
+        .0;
+
+        colored::control::unset_override();
```
why do we need this? do tests not print in color? I would rather not have these lines, but if they're necessary, let's add an explanatory comment and move this into `text_with_carats_and_line_count_buffer_and_line_numbers`

rbalicki2 commented on crates/common_lang_types/src/text_with_carats.rs:
```diff
@@ -116,19 +114,35 @@ fn text_with_carats_and_line_count_buffer_and_line_numbers(
+            output_lines.push(colored_source);
             // a line may be entirely empty, due to containing only a \n. We probably want to avoid
             // printing an empty line underneath. This is weird and probably buggy!
+            let end_of_carats = end_idx as u32;
```
instead of having two variables (`end_of_carats` and `end_idx`) which have the same value but different representations, can we have a single one, and just cast it as needed?

e.g. if `start_of_carats` became `let start_of_carats = (actual_span.start as usize).saturating_sub(start_of_line);` we probably could do with fewer variables and less casting.

I'm not concerned with the perf implications of usize vs u32 here, I just want this code to be understandable later

(and despite my previous comment about multiple PRs, feel free to make this change in this same PR)

rbalicki2 commented on crates/common_lang_types/src/text_with_carats.rs:
```diff
@@ -116,19 +114,35 @@ fn text_with_carats_and_line_count_buffer_and_line_numbers(
-            let start_of_carats = actual_span.start.saturating_sub(start_of_line as u32);
             let line_len = line_content.len() as u32;
-            let end_of_carats = std::cmp::min(
+            let start_of_carats = actual_span.start.saturating_sub(start_of_line as u32);
```
nit: no worries here, but it's slightly easier to review if you don't re-order lines like this. Before submitting PRs, it's a good practice to do `git diff main` and see where there are minor changes like this that add visual noise but don't materially affect the PR. (Likewise, if there are refactors that you can extract, that's good to make as a separate PR, but so far I see none of those.)

rbalicki2 commented on crates/common_lang_types/src/text_with_carats.rs:
```diff
@@ -589,4 +605,21 @@ mod test {
+        colored::control::unset_override();
+
+        assert!(output.contains("\u{1b}[91m"));
+    }
```
could we also do this for at least one test that spans across multiple lines? (Which looks like it works correctly!!! 🎉 )

<img width="692" height="448" alt="Image" src="https://github.com/user-attachments/assets/41bd91d4-585e-47ba-8ba8-81c73e53a8c5" />

rbalicki2 reviewed (commented):
Amazing!!!! 🙌  Thank you so much for doing this!!!

Just a few nits + let's add a few tests for the multiline case, and we're g2g. This is soooo nice!!!!

kartikmawani commented:
Thanks for the detailed review! I've addressed all the points discussed

  I  refactored the loop to calculate start_of_carats and end_of_carats directly as usize. This removed the redundant end_idx variable and the back-and-forth casting between u32 and usize.

Switched to using .normal() for the uncolored branches to match the ColoredString type returned by .bright_red().  

I moved colored::control::unset_override(); inside the main function so it doesn't need to be called in tests.

 Added the requested multiline test case  to verify cross-line highlighting.
Switched all tests to use strict assert_eq! with the exact expected output strings, rather than just checking .contains().

 noted on the "visual noise" regarding line ordering.I'll keep that in mind for future PRs to keep the diffs cleaner
